src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,326 @@
+# % key: l
+# % description: List the search result without downloading
+# %end
+
+# %flag
+# % key: e
+# % description: Extract the downloaded the datasets
+# %end
+
+# %flag
+# % key: d
+# % description: Delete the product archieve after downloading
+# %end
+
+
+import sys
+import os
+import getpass
+from datetime import *
+import grass.script as gs
+from grass.exceptions import ParameterError
+
+
+def create_dir(dir):
+    if not os.path.isdir(dir):
+        try:
+            os.makedirs(dir)
+            return 0
+        except Exception as e:
+            gs.warning(_("Could not create directory {}").format(dir))
+            return 1
+    gs.verbose(_("Directory {} already exists").format(dir))
+    return 0
+
+
+def get_bb(vector=None):
+    args = {}
+    if vector:
+        args["vector"] = vector
+    # are we in LatLong location?
+    kv = gs.parse_command("g.proj", flags="j")
+    if "+proj" not in kv:
+        gs.fatal("Unable to get bounding box: unprojected location not supported")
+    if kv["+proj"] != "longlat":
+        info = gs.parse_command("g.region", flags="uplg", **args)
+        return {
+            "lonmin": info["nw_long"],
+            "latmin": info["sw_lat"],
+            "lonmax": info["ne_long"],
+            "latmax": info["nw_lat"],
+        }
+    info = gs.parse_command("g.region", flags="upg", **args)
+    return {
+        "lonmin": info["w"],
+        "latmin": info["s"],
+        "lonmax": info["e"],
+        "latmax": info["n"],
+    }
+
+
+def download_products(products_list):
+    dag.download_all(products_list)
+
+
+def download_by_id(query_id: str):
+    gs.verbose(
+        _(f"Searching for product ending with: {query_id[-min(len(query_id), 8):]}")
+    )
+    product, count = dag.search(id=query_id)
+    if count != 1:
+        raise ParameterError("Product couldn't be uniquely identified")
+    if not product[0].properties["id"].startswith(query_id):
+        raise ParameterError("Product wasn't found")
+    gs.verbose(_(f"Poduct ending with: {query_id[-min(len(query_id), 8):]} is found."))
+    gs.verbose(_("Downloading..."))
+    dag.download(product[0])
+
+
+def download_by_ids(ids: list):
+    for id in ids:
+        try:
+            download_by_id(id)
+        except ParameterError:
+            gs.error(
+                _(f"Product ending with: {id[-min(len(id), 8):]}, failed to download")
+            )
+
+
+def setup_environment_variables():
+    os.environ[f"EODAG__{options['provider']}__DOWNLOAD__EXTRACT"] = str(flags["e"])
+    os.environ[f"EODAG__{options['provider']}__DOWNLOAD__DELETE_ARCHIV"] = str(
+        flags["d"]
+    )
+
+    if options["output"]:
+        os.environ[f"EODAG__{options['provider']}__DOWNLOAD__OUTPUTS_PREFIX"] = options[
+            "output"
+        ]
+
+
+def main():
+    # products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
+
+    # TODO: Add option for setting a differnt config file path
+    global options, flags
```

Comment:
no need for global, if you put the following line (`options, flags = gs.parser()`) into the `if __name__ == "__main__":` section